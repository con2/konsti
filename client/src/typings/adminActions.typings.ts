import { Game } from 'typings/game.typings';

export const SUBMIT_UPDATE_HIDDEN = 'SUBMIT_UPDATE_HIDDEN';
export const SUBMIT_GET_SETTINGS = 'SUBMIT_GET_SETTINGS';
export const SUBMIT_SELECT_SIGNUP_TIME = 'SUBMIT_SELECT_SIGNUP_TIME';
export const SUBMIT_SET_TEST_TIME = 'SUBMIT_SET_TEST_TIME';
export const SUBMIT_TOGGLE_APP_OPEN = 'SUBMIT_TOGGLE_APP_OPEN';
export const SUBMIT_RESPONSE_MESSAGE = 'SUBMIT_RESPONSE_MESSAGE';

export interface SubmitUpdateHiddenAsync {
  type: typeof SUBMIT_UPDATE_HIDDEN;
  hiddenGames: readonly Game[];
}

export interface SubmitGetSettingsAsync {
  type: typeof SUBMIT_GET_SETTINGS;
  hiddenGames: readonly Game[];
  signupTime: string;
  appOpen: boolean;
}

export interface SubmitSignupTimeAsync {
  type: typeof SUBMIT_SELECT_SIGNUP_TIME;
  signupTime: string;
}

export interface SubmitSetTestTime {
  type: typeof SUBMIT_SET_TEST_TIME;
  testTime: string;
}

export interface SubmitToggleAppOpenAsync {
  type: typeof SUBMIT_TOGGLE_APP_OPEN;
  appOpen: boolean;
}

export interface SubmitResponseMessageAsync {
  type: typeof SUBMIT_RESPONSE_MESSAGE;
  responseMessage: string;
}

export type AdminActionTypes =
  | SubmitUpdateHiddenAsync
  | SubmitGetSettingsAsync
  | SubmitSignupTimeAsync
  | SubmitSetTestTime
  | SubmitToggleAppOpenAsync
  | SubmitResponseMessageAsync;
