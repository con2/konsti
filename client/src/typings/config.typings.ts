export interface Config {
  appName: string;
  SIGNUP_END_TIME: number;
  SIGNUP_OPEN_TIME: number;
  MESSAGE_DELAY: number;
  CONVENTION_NAME: ConventionName;
  CONVENTION_YEAR: string;
  CONVENTION_START_TIME: string;
  DAY_START_TIME: number;
  apiServerURL: string;
  useTestTime: boolean;
  reduxTrace: boolean;
  loadedSettings: string;
  enableAxe: boolean;
  enableWhyDidYouRender: boolean;
  dataUpdateInterval: number;
  noSignupGames: string[];
  revolvingDoorEnabled: boolean;
  tagFilteringEnabled: boolean;
  simpleDetails: boolean;
}

type ConventionName = 'Ropecon' | 'Tracon Hitpoint';
