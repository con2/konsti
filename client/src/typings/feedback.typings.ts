export interface Feedback {
  feedback: string;
  gameId: string;
}

export interface PostFeedbackResponse {
  message: string;
  status: 'success';
}
