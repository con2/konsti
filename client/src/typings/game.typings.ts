export interface Game {
  gameId: string;
  title: string;
  description: string;
  location: string;
  startTime: string;
  mins: number;
  tags: readonly string[];
  genres: readonly string[];
  styles: readonly string[];
  language: string;
  endTime: string;
  people: string;
  minAttendance: number;
  maxAttendance: number;
  gameSystem: string;
  englishOk: boolean;
  childrenFriendly: boolean;
  ageRestricted: boolean;
  beginnerFriendly: boolean;
  intendedForExperiencedParticipants: boolean;
  shortDescription: string;
  revolvingDoor: boolean;
  popularity: number;
  programType: string;
}

export interface DnDUpdatedPositions {
  availableGames?: readonly Game[];
  selectedGames?: readonly Game[];
}

export interface DnDMove {
  index: number;
  droppableId: string;
}

export interface PostGamesUpdateResponse {
  games: Game[];
  message: string;
  status: 'success';
}

export interface GetGamesResponse {
  games: Game[];
  message: string;
  status: 'success';
}

export interface PostHiddenResponse {
  hiddenGames: Game[];
  message: string;
  status: 'success';
}
