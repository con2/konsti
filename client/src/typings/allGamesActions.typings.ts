import { Game } from 'typings/game.typings';

export const SUBMIT_GET_GAMES = 'SUBMIT_GET_GAMES';

export interface SubmitGetGamesAsync {
  type: typeof SUBMIT_GET_GAMES;
  games: readonly Game[];
}

export type AllGamesActionTypes = SubmitGetGamesAsync;
