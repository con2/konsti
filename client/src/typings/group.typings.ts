import { Signup } from 'typings/user.typings';

export interface GroupMember {
  enteredGames: readonly Signup[];
  groupCode: string;
  serial: string;
  signedGames: readonly Signup[];
  username: string;
}

export interface GroupData {
  groupCode: string;
  leader: boolean;
  ownSerial: string;
  username: string;
  leaveGroup?: boolean;
  closeGroup?: boolean;
}

export interface PostGroupResponse {
  groupCode: string;
  message: string;
  status: 'success';
}

export interface GetGroupResponse {
  message: string;
  results: GroupMember[];
  status: 'success';
}
