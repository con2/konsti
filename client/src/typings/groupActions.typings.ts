import { GroupMember } from 'typings/group.typings';

export const SUBMIT_UPDATE_GROUP_CODE = 'SUBMIT_UPDATE_GROUP_CODE';
export const SUBMIT_LEAVE_GROUP = 'SUBMIT_LEAVE_GROUP';
export const SUBMIT_UPDATE_GROUP_MEMBERS = 'SUBMIT_UPDATE_GROUP_MEMBERS';

export interface SubmitUpdateGroupCodeAsync {
  type: typeof SUBMIT_UPDATE_GROUP_CODE;
  groupCode: string;
}

export interface SubmitGetGroupAsync {
  type: typeof SUBMIT_UPDATE_GROUP_MEMBERS;
  groupMembers: readonly GroupMember[];
}

export interface SubmitLeaveGroupAsync {
  type: typeof SUBMIT_LEAVE_GROUP;
  groupCode: string;
}

export type GroupActionTypes =
  | SubmitUpdateGroupCodeAsync
  | SubmitGetGroupAsync
  | SubmitLeaveGroupAsync;
