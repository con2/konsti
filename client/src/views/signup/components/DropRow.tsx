import React, { FC, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';
import { Droppable, Draggable } from 'react-beautiful-dnd';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { Game } from 'typings/game.typings';

export interface Props {
  droppableId: string;
  games: readonly Game[];
  label: string;
  showCount: boolean;
}

export const DropRow: FC<Props> = (props: Props): ReactElement => {
  const { droppableId, games, label, showCount } = props;
  const { t } = useTranslation();

  const getListStyle = (dragging: boolean): string => {
    if (dragging) return 'dragging';
    else return '';
  };

  const getPopularity = (game: Game): string => {
    if (game.popularity >= game.maxAttendance) return 'high-popularity';
    else if (
      game.popularity >= game.maxAttendance / 2 &&
      game.popularity >= game.minAttendance
    )
      return 'medium-popularity';
    else return 'low-popularity';
  };

  return (
    <>
      <p>
        {label} {showCount && <span>({games.length}/3)</span>}
      </p>

      <Droppable droppableId={droppableId}>
        {(provided, snapshot) => (
          <DropRowContainer
            className={`drop-row ${droppableId} ${getListStyle(
              snapshot.isDraggingOver
            )}`}
            ref={provided.innerRef}
          >
            {games.map((game, index) => (
              <Draggable
                key={game.gameId}
                draggableId={game.gameId}
                index={index}
              >
                {(provided) => (
                  <Link to={`/games/${game.gameId}`}>
                    <DraggableItem
                      className={`${getPopularity(game)}`}
                      ref={provided.innerRef}
                      {...provided.draggableProps}
                      {...provided.dragHandleProps}
                    >
                      <SignupGameTitle className='break-long'>
                        {game.title}
                      </SignupGameTitle>
                      <SignupShortDescription className='break-long'>
                        {t(`programType.${game.programType}`)}:{' '}
                        {game.shortDescription
                          ? game.shortDescription
                          : game.gameSystem}
                      </SignupShortDescription>
                      <SignupPopularity>
                        {t('playerStatus')}: {t(getPopularity(game))}
                      </SignupPopularity>
                    </DraggableItem>
                  </Link>
                )}
              </Draggable>
            ))}
            {provided.placeholder}
          </DropRowContainer>
        )}
      </Droppable>
    </>
  );
};

const DropRowContainer = styled.div`
  background-color: ${(props) => props.theme.backgroundHighlight};
  min-height: 40px;
  padding: 10px;

  &.dragging {
    background-color: ${(props) => props.theme.backgroundDndRow};
  }

  &.availableGames,
  &.selectedGames {
    min-height: 500px;
  }

  a {
    color: ${(props) => props.theme.mainText};
    text-decoration: none;
    cursor: grabbing;
  }
`;

const DraggableItem = styled.div`
  background-color: ${(props) => props.theme.backgroundDndItem};
  border: 1px solid ${(props) => props.theme.borderInactive};
  margin: 8px 0;
  padding: 8px;
  box-shadow: 0 1px 0 rgba(9, 30, 66, 0.25);
  border-radius: 3px;

  :hover,
  :focus {
    /* stylelint-disable-next-line plugin/no-unsupported-browser-features */
    filter: brightness(90%);
  }

  &.high-popularity {
    background-color: #ffe8e8;
  }

  &.medium-popularity {
    background-color: #fff;
  }

  &.low-popularity {
    background-color: #f0ffff;
  }
`;

const SignupGameTitle = styled.p`
  margin: 0 0 4px 0;
`;

const SignupShortDescription = styled.p`
  margin: 0 0 0 14px;
  font-size: ${(props) => props.theme.fontSizeSmall};
  font-style: italic;

  @media (max-width: ${(props) => props.theme.breakpointPhone}) {
    margin: 0;
  }
`;

const SignupPopularity = styled.p`
  font-size: ${(props) => props.theme.fontSizeSmall};
  margin-bottom: 0;
`;
