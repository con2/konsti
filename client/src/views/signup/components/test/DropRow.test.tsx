import React from 'react';
import { shallow } from 'enzyme';
import { DropRow, Props } from '../DropRow';
import { Game } from 'typings/game.typings';

const droppableId = 'testId';
const games: Game[] = [];
const label = 'test label';
const showCount = true;

describe('DropRow', () => {
  it('should render correctly', () => {
    const props: Props = {
      droppableId,
      games,
      label,
      showCount,
    };
    const component = shallow(<DropRow {...props} />);
    expect(component).toMatchSnapshot();
  });
});
