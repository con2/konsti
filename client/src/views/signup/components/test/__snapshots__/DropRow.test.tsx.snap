// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropRow should render correctly 1`] = `
<Fragment>
  <p>
    test label
     
    <span>
      (
      0
      /3)
    </span>
  </p>
  <Connect(Droppable)
    direction="vertical"
    droppableId="testId"
    getContainerForClone={[Function]}
    ignoreContainerClipping={false}
    isCombineEnabled={false}
    isDropDisabled={false}
    mode="standard"
    renderClone={null}
    type="DEFAULT"
  >
    <Component />
  </Connect(Droppable)>
</Fragment>
`;
